.github/workflows/ci.yml

```diff
@@ -249,6 +249,34 @@ jobs:
         with:
           files: ./coverage.txt
 
+  vendor:
+    if: github.event_name != 'pull_request'
+    runs-on: ubuntu-latest
+    steps:
+      -
+        name: Checkout
+        uses: actions/checkout@v3
+      -
+        name: Set up Go
+        uses: actions/setup-go@v3
+        with:
+          go-version-file: 'go.mod'
+          check-latest: true
+          cache: true
+      -
+        name: Build vendor
+        run: |
+          go mod vendor
+          mkdir -p bin/release
+          tar czf bin/release/docker-compose-vendor.tar.gz vendor
+      -
+        name: Upload artifacts
+        uses: actions/upload-artifact@v3
```

Comment:
Is there a reason that we use `actions/upload-artifact@v3` instead of `v4`? According to the [action README](https://github.com/actions/upload-artifact?tab=readme-ov-file#actionsupload-artifact), `v3` will be deprecated in November.

Reply:
Speaking of `v4`, some of the breaking changed in it relate to my other comment bellow about the aritfact name.

```
2. Uploading to the same named Artifact multiple times.

Due to how Artifacts are created in this new version, it is no longer possible to upload to the same named Artifact multiple times. You must either split the uploads into multiple Artifacts with different names, or only upload once. Otherwise you will encounter an error.
```

Reply:
aritfact name must be unique for a built commit, I don't think this has any impact for this workflow.

We still use upload-artifact@v3 on this repo as this breaking change require the whole workflow to be revisited to support the new behavior

Reply:
> aritfact name must be unique for a built commit, I don't think this has any impact for this workflow.

Ah ok, then this should be fine.